lib/MongoDB/Database.pm

```diff
@@ -225,23 +230,35 @@ L<listCollections command
 documentation|http://docs.mongodb.org/manual/reference/command/listCollections/>
 for more details on filtering for specific collections.
 
+A hashref of options may also be provided.
+
+Valid options include:
+
+=for :list
+* C<session> - the session to use for these operations. If not supplied, will
+  use an implicit session. For more information see L<MongoDB::ClientSession>
+
 B<Warning:> if the number of collections is very large, this may return
 a very large result.  Either pass an appropriate filter, or use
 L</list_collections> to iterate over collections instead.
 
 =cut
 
+# TODO why does this not call list_collections and iterate through that?
```

Comment:
No idea.

Reply:
Have refactored that down to just work with the return value of list_collections